gdal/gcore/overview.cpp

```diff
@@ -619,9 +636,18 @@ GDALResampleChunk32R_AverageT( double dfXRatioDstToSrc,
                         if( nVal >= 0 && nVal < nEntryCount &&
                             aEntries[nVal].c4 )
                         {
-                            nTotalR += aEntries[nVal].c1;
-                            nTotalG += aEntries[nVal].c2;
-                            nTotalB += aEntries[nVal].c3;
+                            if(bQuadraticAverage)
+                            {
+                                nTotalR += aEntries[nVal].c1*aEntries[nVal].c1;
```

Comment:
`aEntries[nVal].c1 * aEntries[nVal].c1;` to keep style consistent within the file